build-system/babel-plugins/babel-plugin-transform-default-assignment/test/fixtures/transform-assertions/should-transform-destructures/output.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+function test() {
```

Comment:
Some questions on this below. But I just noticed one interesting thing with Closure. As #29293 describes, this will fail:

```
/**
 * @typedef {num: (number|undefined)}
 */
var TypeDef;

/** @param {!TypeDef} props */
function Comp({num = 11}) {
  // num is still `|undefined` here.
}
```

However, this seems to work perfectly fine!?

```
/** @param {!TypeDef} props */
function Comp(props) {
  const {num = 11} = props;
  // num is NO LONGER `|undefined` here.
}
```

No idea what gives. But if true, that maybe:
1. We don't need to rewrite all destructurings - only the arg ones.
2. We can simplify arg rewrites to simply do: `f({x}) {}` -> `f(_arg1) {let {x} = _arg1}`.

Would this work? I think we would minimize our rewrites this way and maybe the `options = {}` issue would also get simpler?

Reply:
Generating the `_arg1` name is difficult, since it has to match the `@param` name. But, limiting this to just param default assignments does make the code easier (and removes one of the weird changes I had to do).